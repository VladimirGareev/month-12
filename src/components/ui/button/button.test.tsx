
import {cleanup, render, screen, fireEvent} from '@testing-library/react'
import { Button } from './button';

describe("Button Testing", ()=>{
    beforeEach(cleanup);

    it ('Кнопка отрисовывается с текстом', ()=>{
        render (<Button text='testing'/>);
        const button = screen.getByTestId("button");
        expect(button).toHaveTextContent('testing');
        expect(button).toMatchSnapshot();
    })
    
    it ('Кнопка отрисовывается без текста', ()=>{
        render (<Button />);
        const button = screen.getByTestId("button");
        expect(button).toHaveTextContent('');
        expect(button).toMatchSnapshot();
    })

    it ('Кнопка disabled', ()=>{
        render (<Button disabled/>);
        const button = screen.getByTestId("button");
        expect(button).toBeDisabled();
        expect(button).toMatchSnapshot();
    })

    it ('Кнопка отрисовывается с loader', ()=>{
        render (<Button isLoader />);
        const loader = screen.getByTestId("loader");
        const button = screen.getByTestId("button");
        expect(button).toContainElement(loader);
        expect(button).toMatchSnapshot();
    })

    it ('Коллбэк работает корректно', ()=>{
        const callback = jest.fn();
        render (<Button onClick={callback}/>);
        const button = screen.getByTestId("button");
        fireEvent.click(button)
        expect(callback).toHaveBeenCalled();
    })

})