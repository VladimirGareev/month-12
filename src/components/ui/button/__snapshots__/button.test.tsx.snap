// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Testing Кнопка disabled 1`] = `
<button
  class="text text_type_button text_color_primary button undefined false "
  data-testid="button"
  disabled=""
  type="button"
>
  <p
    class="text undefined"
  />
</button>
`;

exports[`Button Testing Кнопка отрисовывается без текста 1`] = `
<button
  class="text text_type_button text_color_primary button undefined false "
  data-testid="button"
  type="button"
>
  <p
    class="text undefined"
  />
</button>
`;

exports[`Button Testing Кнопка отрисовывается с loader 1`] = `
<button
  class="text text_type_button text_color_primary button undefined loader "
  data-testid="button"
  disabled=""
  type="button"
>
  <img
    alt="Загрузка."
    class="loader_icon"
    data-testid="loader"
    src="loader.svg"
  />
</button>
`;

exports[`Button Testing Кнопка отрисовывается с текстом 1`] = `
<button
  class="text text_type_button text_color_primary button undefined false "
  data-testid="button"
  type="button"
>
  <p
    class="text undefined"
  >
    testing
  </p>
</button>
`;
